engine/access/state_stream/streamer.go

```diff
@@ -25,21 +26,30 @@ type Streamable interface {
 // Streamer
 type Streamer struct {
 	log         zerolog.Logger
+	sub         Streamable
 	broadcaster *engine.Broadcaster
 	sendTimeout time.Duration
-	sub         Streamable
+	limiter     *rate.Limiter
 }
 
 func NewStreamer(
 	log zerolog.Logger,
 	broadcaster *engine.Broadcaster,
 	sendTimeout time.Duration,
+	limit float64,
 	sub Streamable,
 ) *Streamer {
+	var limiter *rate.Limiter
+	if limit > 0 {
```

Comment:
if limit == 0, can we still create a limiter with a much much higher limit? So that `limiter` is never `nil`?

Reply:
I opted to skip the limiter entirely to avoid unnecessary CPU cycles going to the rate limit calculation. It would simplify the code a bit to use an unlimited limiter, but not enough to justify the overhead imo.